### PR TITLE
invoices: ensure ntfns are sent in the same order

### PR DESCRIPTION
**Pull request by cfromknecht**
SettleHodlInvoice and CancelInvoice both notifyClients after
notifyHodlSubscribers. This commit changes UpdateInvoice to follow the
same pattern so that we are consistent.

### CODE REVIEW

**Comment by cfromknecht** [comment]
yep, only thing i can think of is that an rpc client could see that an htlc was settled before the link became aware of it. pretty unlikely that the difference is reliable enough to be dependent on tho

File: invoices/invoiceregistry.go [hunk 1]
```diff
@@ -846,7 +846,7 @@ func (i *InvoiceRegistry) notifyExitHopHtlcLocked(
 	}
 
 	if updateSubscribers {
-		i.notifyClients(ctx.hash, invoice, invoice.State)
+		defer i.notifyClients(ctx.hash, invoice, invoice.State)
```
Comment (joostjager) [inline invoices/invoiceregistry.go:849]: Can the timing of this call be more explicit? Defer for clean up is one thing, but using defer to run kick off notifications may be too much. Suggestion: don't return in the switch cases but do at at the very end of the function. Right before, notify clients.

Comment (cfromknecht) [reply invoices/invoiceregistry.go:849]: this is pulled out from a much larger pr which i'd rather not rebase for cosmetics. i guess we can just park this PR and circle back once those changes are in, it's not super urgent

**Comment by cfromknecht** [comment]
@joostjager ready for another pass. turned out to be much less code than expected

**Pull request merged**